### PR TITLE
pin `roman_datamodels>=0.22.0` and `stcal>=1.10.0`, and remove `rad` pin

### PR DESCRIPTION
**Pull request by zacharyburnett**
## Tasks
- [ ] **request a review from someone specific**, to avoid making the maintainers review every PR
- [x] add a build milestone, i.e. `24Q4_B15` (use the [latest build](https://github.com/spacetelescope/romancal/milestones) if not sure)
- [ ] Does this PR change user-facing code / API? (if not, label with `no-changelog-entry-needed`)
  - [ ] write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see below for change types)
  - [ ] update or add relevant tests
  - [ ] update relevant docstrings and / or `docs/` page
  - [ ] [start a regression test](https://github.com/spacetelescope/RegressionTests/actions/workflows/romancal.yml) and include a link to the running job ([click here for instructions](https://github.com/spacetelescope/RegressionTests/blob/main/docs/running_regression_tests.md))
    - [ ] Do truth files need to be updated ("okified")?
      - [ ] **after the reviewer has approved these changes**, run `okify_regtests` to update the truth files
- [ ] if a JIRA ticket exists, [make sure it is resolved properly](https://github.com/spacetelescope/romancal/wiki/How-to-resolve-JIRA-issues)

<details><summary>news fragment change types...</summary>

  - ``changes/<PR#>.general.rst``: infrastructure or miscellaneous change
  - ``changes/<PR#>.docs.rst``
  - ``changes/<PR#>.stpipe.rst``
  - ``changes/<PR#>.associations.rst``
  - ``changes/<PR#>.scripts.rst``
  - ``changes/<PR#>.mosaic_pipeline.rst``
  - ``changes/<PR#>.patch_match.rst``

  ## steps
  - ``changes/<PR#>.dq_init.rst``
  - ``changes/<PR#>.saturation.rst``
  - ``changes/<PR#>.refpix.rst``
  - ``changes/<PR#>.linearity.rst``
  - ``changes/<PR#>.dark_current.rst``
  - ``changes/<PR#>.jump_detection.rst``
  - ``changes/<PR#>.ramp_fitting.rst``
  - ``changes/<PR#>.assign_wcs.rst``
  - ``changes/<PR#>.flatfield.rst``
  - ``changes/<PR#>.photom.rst``
  - ``changes/<PR#>.flux.rst``
  - ``changes/<PR#>.source_detection.rst``
  - ``changes/<PR#>.tweakreg.rst``
  - ``changes/<PR#>.skymatch.rst``
  - ``changes/<PR#>.outlier_detection.rst``
  - ``changes/<PR#>.resample.rst``
  - ``changes/<PR#>.source_catalog.rst``
</details>

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -23,13 +23,11 @@ dependencies = [
...
-    # "rad>=0.21.0,<0.22.0",
-    "rad @ git+https://github.com/spacetelescope/rad.git",
-    # "roman_datamodels>=0.21.0,<0.22.0",
-    "roman_datamodels @ git+https://github.com/spacetelescope/roman_datamodels.git",
+    "roman_datamodels>=0.22.0,<0.23.0",
+    # "roman_datamodels @ git+https://github.com/spacetelescope/roman_datamodels.git",
     "scipy >=1.11",
-#    "stcal>=1.8.0,<1.9.0",
-    "stcal @ git+https://github.com/spacetelescope/stcal.git@main",
+   "stcal>=1.9.0,<1.10.0",
```
Comment (braingram) [inline pyproject.toml:29]:
```suggestion
   "stcal>=1.10.0,<1.11.0",
```
Look like we need a stcal release also to support https://github.com/spacetelescope/romancal/pull/1395

Would you make a 1.10.0 release for stcal?

Comment (zacharyburnett) [reply pyproject.toml:29]: on it!

**Comment by braingram** [comment]
There was some discussion of adding upper pins for other dependencies (drizzle, tweakwcs, etc).

drizzle 2.0.0 is already out (and producing [warnings](https://github.com/spacetelescope/romancal/actions/runs/11846176829/job/33013236630#step:10:262) in the test runs) so I'm not sure if there's much of a benefit to adding a pin.

I'd say let's not try something new by adding a bunch of extra pins (beyond what's already in this PR) for this release but perhaps discuss it for the next one. @schlafly how does that sound to you?

**Review by schlafly — APPROVED** [review]
You and Brett know best; this looks good to me.

**Comment by zacharyburnett** [comment]
> One more suggestion. I vote for adding a `<3.13` here:
> 
> https://github.com/spacetelescope/romancal/blob/14d5f82744ed29c192ee075da659948d33ef6723/pyproject.toml#L5
> 
> 
> for the release since scikit image has not yet released a 3.13 compatible version (so we don't have a 3.13 compatible stcal, etc).

`stcal` already has a pin for `<3.13` so that should propagate here

**Pull request merged**